swap_meet/vendor.py

```diff
@@ -1,2 +1,113 @@
 class Vendor:
-    pass
\ No newline at end of file
+    """
+    A class to represent a vendor, managing the inventory and providing
+    methods for adding, removing, and swapping items between vendors.
+    """
+
+    def __init__(self, inventory=None):
+        """
+        Initializes a Vendor instance with an optional inventory list.
+        If no inventory is provided, an empty list is initialized.
+        """
+        self.inventory = inventory or []
+
+    def add(self, new_item_added):
+        """
+        Adds the item to the `inventory`
+        and returns item
+        """
+        self.inventory.append(new_item_added)
+        return new_item_added
+
+    def remove(self, item_to_remove):
+        """
+        Removes the specified item from the `inventory` if it exists and returns item
+        """
+        if item_to_remove not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item_to_remove)
+            return item_to_remove
+
+    def get_by_id(self, item_id):
+        """
+        Check if unique item id matches an item id in inventory and return item
+        """
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        """
+        Swaps the specified items between this vendor and another vendor,
+        if both vendors have the specified items in their inventory.
+        """
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item)
+            other_vendor.inventory.append(my_item)
+            other_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+        return False
+
+    def swap_first_item(self, other_vendor):
+        """
+        Swaps the first item from this vendor's inventory with the first item
+        from another vendor's inventory.
+        """
+        if len(self.inventory) and len(other_vendor.inventory):
+            my_first_item = self.inventory[0]
+            their_first_item = other_vendor.inventory[0]
+            return self.swap_items(other_vendor, my_first_item, their_first_item)
+
```

Comment:
If input is invalid then this method will implicitly return None since there is no return statement outside of the conditional block. Per the README this function should return False if input is invalid.